bson/primitive/primitive_test.go

```diff
@@ -199,3 +200,334 @@ func TestDateTime(t *testing.T) {
 		})
 	})
 }
+
+func TestMarshalingD(t *testing.T) {
```

Comment:
Optional: Consider a more descriptive test name (borrowed from testable examples name conventions).

```suggestion
func TestD_MarshalJSON(t *testing.T) {
```